components/brave_sync/brave_sync_service_impl.cc

```diff
@@ -182,6 +182,8 @@ void BraveSyncServiceImpl::OnSetupSyncNewToSync(
     return;
   }
 
+  sync_words_.clear();  // If the previous attempt was connect to sync chain
```

Comment:
can you also add "and failed to receive `save-init-data`"?

Reply:
thanks, this is done